ci/jenkins/nephe.sh

```diff
@@ -0,0 +1,184 @@
+        --vchost                VC ipAddress or domain name to deploy vm.
+        --vcuser                User name for VC.
+        --goVcPassword          Password to the user name for VC.
+        --datacenter            Data center that is used to deploy vm.
+        --datastore             Data store that is used to deploy vm.
+        --vcCluster             VC cluster that is used to deploy vm.
+        --resourcePool          Resource pool that is used to deploy vm.
+        --vcNetwork             Network that is used to deploy vm.
+        --virtualMachine        VM template that is used to deploy vm.
+        --testType              The type of tests that will be triggered."
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --buildnumber)
+    buildNumber="$2"
+    shift 2
+    ;;
+    --vchost)
+    vcHost="$2"
+    shift 2
+    ;;
+    --vcuser)
+    vcUser="$2"
+    shift 2
+    ;;
+    --datacenter)
+    dataCenterName="$2"
+    shift 2
+    ;;
+    --datastore)
+    dataStore="$2"
+    shift 2
+    ;;
+    --vcCluster)
+    vcCluster="$2"
+    shift 2
+    ;;
+    --vcNetwork)
+    vcNetwork="$2"
+    shift 2
+    ;;
+    --resourcePool)
+    resourcePoolPath="$2"
+    shift 2
+    ;;
+    --virtualMachine)
+    virtualMachine="$2"
+    shift 2
+    ;;
+    --goVcPassword)
+    goVcPassword="$2"
+    shift 2
+    ;;
+    --testType)
+    testType="$2"
+    shift 2
+    ;;
+esac
+done
+
+OLDPWD=`pwd`
+
+cd ci/jenkins
+if [ ! -e id_rsa ]; then
+  ssh-keygen -t rsa -P '' -f id_rsa
+fi
+if [ ! -e playbook/jenkins_id_rsa ];then
+  ssh-keygen -t rsa -P '' -f playbook/jenkins_id_rsa
+fi
+chmod 0600 id_rsa
+chmod 0600 playbook/jenkins_id_rsa
+testbed_name="nephe-test-${testType}-${buildNumber}"
+
+sudo apt install -y unzip ansible
+
+cat terraform-${vcHost}.tfvars
+rm -rf terraform-${vcHost}.tfvars
+cat << EOF > terraform-${vcHost}.tfvars
+vsphere_user="${vcUser}"
+vm_count=1
+vsphere_datacenter="${dataCenterName}"
+vsphere_datastore="${dataStore}"
+vsphere_compute_cluster="${vcCluster}"
+vsphere_resource_pool="${resourcePoolPath}"
+vsphere_network="${vcNetwork}"
+vsphere_virtual_machine="${virtualMachine}"
+EOF
+cat terraform-${vcHost}.tfvars
+
+./deploy.sh ${testbed_name} ${vcHost} ${goVcPassword}
+
+ip_addr=`cat terraform.tfstate.d/${testbed_name}/terraform.tfstate | jq -r .outputs.vm_ips.value[0]`
+chmod 0600 id_rsa
+
+ssh -i id_rsa ubuntu@${ip_addr} "sudo apt-get update -y && sudo apt-get install -y ca-certificates curl unzip gnupg lsb-release"
+#TODO: Scp'ing the code. Need to find better way
+#scp -r -i id_rsa ${OLDPWD}/* ubuntu@${ip_addr}:~/
+# clone code from github repo, and checkout commit hash same as on jenkins node
+pr_number=`echo ${sha1}|cut -d '/' -f3`
```

Comment:
dont think this is a good idea.. We wont be able to run nightly test on master branch

Reply:
@lzhecheng for inputs too